### PR TITLE
cmake: x86 and xtensa toolchain updates for SDK 0.11

### PR DESCRIPTION
**Pull request by dcpleung**
This includes the cmake changes needed for x86 and xtensa toolchains in SDK 0.11.

Includes:
* Using x86_64 toolchain to build 32-bit x86 targets, so there is no need for a separate x86 32-bit toolchain (i586-zephyr-elf) anymore.
* On Xtensa, each SoC requires its own toolchain. In order to distinguish among them, each Xtensa toolchain is placed in it own directory under the SoC name, allowing the build system to utilize different toolchain for different Xtensa SoCs.
* Due to the Xtensa toolchain change and the addition of toolchain for `intel_s1000_crb` board, it is now possible to build for this board using Zephyr SDK. Changes are included to make this work, and these will allow us to build in CI.

### CODE REVIEW

File: cmake/toolchain/xtools/generic.cmake [hunk 1]
```diff
@@ -15,15 +15,24 @@ set(BINTOOLS gnu)
 # should use. Note that we can't use ARCH to distinguish between
 # toolchains because choosing between iamcu and non-iamcu is dependent
 # on Kconfig, not ARCH.
+if("${ARCH}" STREQUAL "xtensa")
+file(GLOB toolchain_paths ${TOOLCHAIN_HOME}/xtensa/*/*)
```
Comment (SebastianBoe) [inline cmake/toolchain/xtools/generic.cmake]: indentation

**Review by tyrantlizard77 — APPROVED** [review]

**Comment by dcpleung** [comment]
FYI, this would not pass CI until we have 0.11 SDK, since the Xtensa sample_controller toolchain cannot compile for intel_s1000.

**Comment by finikorg** [comment]
@dcpleung I am trying to use `sys_read32()` functions and the headers include at some point header `include/toolchain/gcc.h` with `GEN_ABSOLUTE_SYM` undefined for the `CONFIG_XTENSA`. Should it be defined also?

**Comment by dcpleung** [comment]
Most of these are merged. The remaining patches can be submitted separately.

**Pull request closed without merging**